lib/matplotlib/tests/test_scale.py

```diff
@@ -55,6 +57,25 @@ def test_symlog_mask_nan():
     assert type(out) == type(x)
 
 
+def test_symlog_linthresh():
+    fig, ax = plt.subplots()
+
+    n_samples = 100
+
+    upper_bound = 1.0
+
+    x = [random.uniform(0.0, upper_bound) for _ in range(n_samples)]
+    y = [random.uniform(0.0, upper_bound) for _ in range(n_samples)]
+
+    with pytest.warns(UserWarning) as record:
+        plt.plot(x, y, 'o')
+        ax.set_xscale('symlog')
+        ax.set_yscale('symlog')
+        plt.show()
+
+    assert len(record) == 1
```

Comment:
The `pytest.warns` context already checks for the existence of the warning. No need to track `record`. To, make the check more targeted, it's good to regex-match the message.

```suggestion
    with pytest.warns(UserWarning, match="All values .* below linthresh"):
        plt.plot(x, y, 'o')
        ax.set_xscale('symlog')
        ax.set_yscale('symlog')
        plt.show()
```

Reply:
Ah, one more thing: Only put the line that generates the warning in the context. This (1) makes the check more narrow and (2) communicates to the reader where we expect the warning to happen.